ons-spark/spark-analysis/flags.ipynb

```diff
@@ -0,0 +1,205 @@
+    "\n",
+    "It is likely that there are *many* ways of doing this. Below is one method using the functions available from `pyspark.sql` and some dummy data. \n",
+    "\n",
+    "As is good practice, first import the packages you will be using and start your Spark session:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pyspark.sql.functions as F\n",
+    "from pyspark.sql import SparkSession, Window"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "spark = SparkSession.builder.appName(\"troubleshooting\").getOrCreate()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Creating your dummy data\n",
+    "\n",
+    "The dataframe `df` will have an `id` column from 0 to 4. To create multiple `id` entries we will do a `.CrossJoin` with the numbers from 0 to 2 and then drop the latter column. To find more information on cross joins please refer to the [page on cross joins](https://best-practice-and-impact.github.io/ons-spark/spark-functions/cross-joins.html). Finally, we will add an `age` column with random numbers from 1 to 10. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "+---+---+\n",
+      "| id|age|\n",
+      "+---+---+\n",
+      "|  0|  7|\n",
+      "|  0|  9|\n",
+      "|  0| 10|\n",
+      "|  1|  9|\n",
+      "|  1|  5|\n",
+      "|  1|  6|\n",
+      "|  2|  4|\n",
+      "|  2|  3|\n",
+      "|  2| 10|\n",
+      "|  3|  1|\n",
+      "|  3|  4|\n",
+      "|  3| 10|\n",
+      "|  4|  8|\n",
+      "|  4|  7|\n",
+      "|  4|  3|\n",
+      "+---+---+\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "df = (\n",
+    "    spark.range(5)\n",
+    "    .crossJoin(\n",
+    "        spark.range(3)\n",
+    "        .withColumnRenamed(\"id\",\"drop\")\n",
+    "    ).drop(\"drop\")\n",
+    "    .withColumn(\"age\", F.ceil(F.rand(seed=42)*10))\n",
+    ")\n",
+    "\n",
+    "df.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Creating your columns \n",
+    "\n",
+    "There will be more than one way to get the desired result. Note that the method outlined here spells out the process in detail; please feel free to combine some of the steps (without sacrificing code readbility of course!).\n",
+    "\n",
+    "Now that we have dummy data we want to create a `window` over `id` that is ordered by `age` in descending order. We then want to create the following columns:\n",
+    "\n",
+    "- `age_lag` showing the next highest age within an id\n",
+    "- `age_diff` obvious enough\n",
```

Comment:
I worry that the "obvious enough" may come across a little patronising and may be better to just explain it (appreciate that it was written this way in the original notebook so definitely not your fault!)